src/density_interface.jl

```diff
@@ -0,0 +1,5 @@
+@inline DensityInterface.hasdensity(::Distribution) = true
+
+DensityInterface.logdensityof(d::Distribution, x) = logpdf(d, x)
```

Comment:
The log likelihood example makes me wonder if this should be

```suggestion
DensityInterface.logdensityof(d::Distribution, x) = loglikelihood(d, x)
```

The difference would be:
- `loglikelihood` is guaranteed to return a `Real` whereas `logpdf` can return an array of `Real`s if `x` consists of multiple samples (in the univariate case though it has to be broadcasted)
- `loglikelihood` evaluates the log likelihood of multiple samples without constructing intermediate arrays
- they coincide for single samples

Due to this more consistent interface and since it does not allocate eg Turing mainly uses `loglikelihood` but not `logpdf` internally.